src/pyodide/internal/python.ts

```diff
@@ -162,22 +170,29 @@ function getEmscriptenSettings(lockfile, indexURL) {
  *
  * Returns the instantiated emscriptenModule object.
  */
-async function instantiateEmscriptenModule(emscriptenSettings) {
+async function instantiateEmscriptenModule(
+  emscriptenSettings: EmscriptenSettings,
+): Promise<Module> {
   try {
     // Force Emscripten to feature detect the way we want
     // They used to have an `environment` setting that did this but it has been
     // removed =(
     // If/when we link our own Pyodide we can remove this.
+    // @ts-ignore
     globalThis.window = {}; // makes ENVIRONMENT_IS_WEB    = true
+    // @ts-ignore
     globalThis.importScripts = 1; // makes ENVIRONMENT_IS_WORKER = false
     const p = _createPyodideModule(emscriptenSettings);
+    // @ts-ignore
     delete globalThis.window;
+    // @ts-ignore
     delete globalThis.importScripts;
     const emscriptenModule = await p;
     return emscriptenModule;
   } catch (e) {
     console.warn("Error in instantiateEmscriptenModule");
     reportError(e);
+    throw e;
```

Comment:
The reportError above already throws, this is redundant.

Reply:
Right, I had to make the return type of reportError `never` so TypeScript would stop complaining about the lack of return